src/test/unit_tests/test_namespace_fs.js

```diff
@@ -426,11 +413,152 @@ mocha.describe('namespace_fs', function() {
 
 });
 
+mocha.describe('nsfs_symlinks_validations', function() {
+
+    let tmp_fs_path = '/tmp/test_nsfs_symboliclinks';
+    if (process.platform === MAC_PLATFORM) {
+        tmp_fs_path = '/private/' + tmp_fs_path;
+    }
+    const bucket = 'bucket1';
+    const bucket_full_path = tmp_fs_path + '/' + bucket;
+    const expected_dirs = ['d1', 'd2', 'd3/d3d1'];
+    const expected_files = ['f1', 'f2', 'f3', 'd2/f4', 'd2/f5', 'd3/d3d1/f6'];
+    const expected_links = [{ t: 'f1', n: 'lf1' }, { t: '/etc', n: 'ld2' }];
+    const dummy_object_sdk = { requesting_account: { nsfs_account_config: { uid: process.getuid(), gid: process.getgid() } } };
+
+    const ns = new NamespaceFS({ bucket_path: bucket_full_path, bucket_id: '1', namespace_resource_id: undefined });
+
+    mocha.before(async () => {
+        await fs_utils.create_fresh_path(`${bucket_full_path}`);
+        await P.all(_.map(expected_dirs, async dir =>
+            fs_utils.create_fresh_path(`${bucket_full_path}/${dir}`)));
+        await P.all(_.map(expected_files, async file =>
+            fs_utils.create_file(`${bucket_full_path}/${file}`)));
```

Comment:
you can use `fs.promises.appendFile` to create a file

Reply:
Done